api/XMLHttpRequest.json

```diff
@@ -2091,6 +2091,66 @@
             "deprecated": false
           }
         }
+      },
+      "worker_support": {
+        "__compat": {
+          "description": "Available in workers",
+          "support": {
+            "chrome": {
+              "version_added": "1",
```

Comment:
I have a hard time believing any of these supported workers before, for example, supporting `Worker()`. Can you re-check all these versions against the worker APIs to make sure we're not suggesting that worker support preceded workers themselves? (e.g., I would expect `api.XMLHttpRequest.worker_support` ≥ `api.Worker`, or some other similar comparison, to hold)

Reply:
Good point - we are - below are the compatibilities from the BCD for workers and shared workers.

Thingy | Chrome | Edge | Firefox | Internet Explorer | Opera | Safari | WebView Android | Chrome Android | Firefox Android | Opera Android | iOS Safari | Samsung Internet
-- |  -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
XMLHttpRequest | 1 | 12 | 1 | 7 | 8 | 1.2 | 1 | 18 | 4 | 10.1 | 1 | 1.0
Worker | 4 | 12 | 3.5 | 10 | 10.6 | 4 | 4 | 18 | 4 | 11 | 5.1 | 1.0
SharedWorker | 4 | 79 | 29 | X | 10.6 | No support 5 — 6.1 | X | X | 33 | No support 11 — 14 | No support 5.1 — 7 | No support4.0 — 5.0

I propose that we set the version to true for all of these and modify the note from:

> "notes": "Not supported in service workers."
 
 TO
 
 > "notes": "Not supported in service workers. Supported in dedicated and shared workers when they are supported by the platform (check Worker and SharedWorker compatibility for details).

Of course this may not be 100% accurate but it does convey that you need to check the Worker/Shared worker. 

What do you think?

Reply:
I think setting `true` values and using the note here muddies things a little too much. In general, let's treat this support as "supported in at least one or more worker types". More concretely:

- Let's set all the versions to whatever the support is on `api.Worker`. This feels like a pretty safe assumption and if we're off by a version or two then it seems pretty harmless for something that happened a decade or more ago. And if we're wrong on the internet, it's likely someone will be along shortly to tell us.
- Let's stick to the original "Not supported in service workers." note. It's hard to deal with `SharedWorker` in a way that doesn't just duplicate `SharedWorker`'s compat data ("From version 1 to 4, `SharedWorker` is unsupported.") or creates some ambiguity (e.g., implying that shared workers exist but don't support this API with a something like "Before version 4, not supported in shared workers."). The edge here shouldn't be too sharp anyway: you'll get an error trying to use a non-existent `SharedWorker`, not `XMLHttpRequest`.

Reply:
Thanks @ddbeck - Makes sense to me. Done!